Dependency/peg-markdown-highlight/HGMarkdownHighlighter.m

```diff
@@ -101,15 +99,18 @@ - (instancetype) initWithTextView:(NSTextView *)textView
 
 #pragma mark -
 
+- (dispatch_queue_t)parsingQueue
+{
+    if (_parsingQueue == nil)
+    _parsingQueue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
```

Comment:
Let’s use a seperate queue then.

```objc
dispatch_queue_create("macdown-highlighter-parse", DISPATCH_QUEUE_SERIAL)
```

would work. This likely wouldn’t affect much performance-wise, but has the marginal benefit of avoiding blocking other operations if the parser fails (or being blocked if another operation somewhere fails).